Building a NET7/NET8 iOS project hangs with LLVM

### Steps to Reproduce

1. Have a `net8.0-android`/`net8.0-ios` project with the following csproj:
```csproj
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<LangVersion>11.0</LangVersion>
		<TargetFrameworks>net8.0-android;net8.0-ios</TargetFrameworks>
		<SingleProject>true</SingleProject>
		<GenerateDocumentationFile>true</GenerateDocumentationFile>
		<OutputType>Exe</OutputType>
		<IsUnoHead>true</IsUnoHead>
		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net8.0-android'">29.0</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net8.0-ios'">14.2</SupportedOSPlatformVersion>
		<NoWarn>Uno0001</NoWarn>
		<FirebaseCrashlyticsUploadSymbolsEnabled>True</FirebaseCrashlyticsUploadSymbolsEnabled>
	</PropertyGroup>
	<!-- Linker configuration. -->
	<PropertyGroup>
		<PublishTrimmed>True</PublishTrimmed>
		<TrimMode>partial</TrimMode>
	</PropertyGroup>
	<ItemGroup>
		<PackageReference Include="Chinook.DataLoader.Uno.WinUI" Version="1.2.0" />
		<PackageReference Include="Chinook.DynamicMvvm.Uno.WinUI" Version="1.1.2" />
		<PackageReference Include="Chinook.SectionsNavigation.Uno.WinUI" Version="2.1.2" />
		<PackageReference Include="ExtendedSplashScreen.Uno.WinUI" Version="1.0.2" />
		<PackageReference Include="MessageDialogService.Uno.WinUI" Version="0.6.0-dev.58" />
		<PackageReference Include="Nventive.View.Uno.WinUI" Version="0.5.0-dev.70" />
		<PackageReference Include="Reactive.Annex.Uno.WinUI" Version="0.6.0-dev.50" />
		<PackageReference Include="ReviewService.NativePrompters" Version="1.0.0" />
		<PackageReference Include="Uno.CommunityToolkit.WinUI" Version="7.1.100" />
		<PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Controls" Version="7.1.100" />
		<PackageReference Include="Uno.Toolkit.WinUI.Material" Version="4.2.22" />
		<PackageReference Include="Uno.WinUI" Version="4.10.37" />
		<PackageReference Include="Uno.WinUI.RemoteControl" Version="4.10.37" Condition="'$(Configuration)'=='Debug'" />
		<PackageReference Include="Uno.WinUI.Lottie" Version="4.10.37" />
		<PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.10.37" />
		<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
		<PackageReference Include="Serilog.Sinks.Xamarin" Version="1.0.0" />
		<PackageReference Include="Serilog.Sinks.File" Version="5.0.0" />
		<PackageReference Include="Xamarin.Build.Download" Version="0.11.4" />
		<PackageReference Include="OneSignalSDK.DotNet" Version="4.3.3" />
		<PackageReference Include="Microsoft.Maui.Essentials" Version="8.0.3" />
		<!-- And many more internal packages that targets net6.0, net6.0-android & net6.0-ios -->
	</ItemGroup>
	<ItemGroup>
		<ProjectReference Include="..\ProjectNa.Access\ProjectName.Access.csproj" />
		<ProjectReference Include="..\ProjectName.Business\ProjectName.Business.csproj" />
		<ProjectReference Include="..\ProjectName.Presentation\ProjectName.Presentation.csproj" />
	</ItemGroup>
	<!-- Disable Bugsee for Production. -->
	<ItemGroup Condition="'$(ApplicationEnvironment)'!='Production'">
		<PackageReference Include="Bugsee" Version="3.0.0" />
	</ItemGroup>
	<ItemGroup>
		<Folder Include="Android\Resources\color\" />
	</ItemGroup>
	<Choose>
		<When Condition="'$(TargetFramework)'=='net8.0-android'">
			<PropertyGroup>
				<AndroidCreatePackagePerAbi>False</AndroidCreatePackagePerAbi>
				<BundleAssemblies>False</BundleAssemblies>
			</PropertyGroup>

			<Choose>
				<When Condition="'$(Configuration)'=='Debug'">
					<PropertyGroup>
						<AndroidUseSharedRuntime>True</AndroidUseSharedRuntime>
						<UseHighDPIResources>False</UseHighDPIResources>
						<EmbedAssembliesIntoApk>False</EmbedAssembliesIntoApk>
						<EnableLLVM>False</EnableLLVM>
						<AndroidEnableProfiledAot>False</AndroidEnableProfiledAot>
						<AndroidDexTool>d8</AndroidDexTool>
						<Optimize>False</Optimize>
						<RunAOTCompilation>False</RunAOTCompilation>
					</PropertyGroup>
				</When>
				<When Condition="'$(Configuration)'=='Release'">
					<PropertyGroup>
						<EmbedAssembliesIntoApk>True</EmbedAssembliesIntoApk>
						<AndroidManagedSymbols>True</AndroidManagedSymbols>
						<AndroidEnableSGenConcurrent>True</AndroidEnableSGenConcurrent>
						<AndroidAotAdditionalArguments>nodebug,no-write-symbols</AndroidAotAdditionalArguments>
						<AndroidUseDefaultAotProfile>False</AndroidUseDefaultAotProfile>
						<AndroidUseAapt2>True</AndroidUseAapt2>
					</PropertyGroup>
					<Choose>
						<When Condition="'$(IsLightBuild)'=='True'">
							<!-- Build Optimization for LightBuild (Faster Builds in Pull Requests). -->
							<PropertyGroup>
								<AndroidEnableProfiledAot>False</AndroidEnableProfiledAot>
								<EnableLLVM>False</EnableLLVM>
								<Optimize>False</Optimize>
								<RunAOTCompilation>False</RunAOTCompilation>
							</PropertyGroup>
						</When>
						<Otherwise>
							<PropertyGroup>
								<!-- 
									Profiled AOT has been disabled, because it causing issues when building on release since targetting NET8. 
									See https://github.com/xamarin/xamarin-android/issues/8739 for more details.
								-->
								<!--<AndroidEnableProfiledAot>True</AndroidEnableProfiledAot>-->
								<!-- LLVM currently doesn't work with AOT enabled and custom profile. -->
								<!-- See https://github.com/dotnet/runtime/issues/74021 for more details. -->
								<!-- See https://github.com/xamarin/xamarin-android/issues/7352 for more details. -->
								<EnableLLVM>False</EnableLLVM>
								<Optimize>True</Optimize>
								<RunAOTCompilation>True</RunAOTCompilation>
							</PropertyGroup>
						</Otherwise>
					</Choose>
				</When>
			</Choose>
			<ItemGroup>
				<!-- And many more internal packages that targets net6.0-android -->
				<PackageReference Include="Uno.UniversalImageLoader" Version="1.9.36" />
				<PackageReference Include="Mono.AotProfiler.Android" Version="7.0.0" />
				<PackageReference Include="Xamarin.Kotlin.StdLib.Jdk8" Version="1.9.20" />
				<PackageReference Include="Xamarin.Firebase.Crashlytics" Version="118.4.3.1" />
				<PackageReference Include="Xamarin.Firebase.Analytics" Version="121.3.0.3" />
				<PackageReference Include="Xamarin.Firebase.Perf" Version="120.5.0" />
				<PackageReference Include="Xamarin.GooglePlayServices.Wallet" Version="119.2.1.1" />
				<PackageReference Include="Xamarin.AndroidX.Lifecycle.LiveData" Version="2.6.1.2" />
				<PackageReference Include="Xamarin.GooglePlayServices.Location" Version="121.0.1.3" />

				<!-- Those two packages are mandatory, otherwise, we can't compile with latest AndroidX packages. See https://github.com/xamarin/AndroidX/issues/764 for more details. -->
				<PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.3.0.1" />
				<PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.8.0.1" />
			</ItemGroup>
			<ItemGroup>
				<!-- 
					Profiled AOT has been disabled, because it causing issues when building on release since targetting NET8. 
					See https://github.com/xamarin/xamarin-android/issues/8739 for more details.
				-->
				<!--<AndroidAotProfile Include="custom.aprof" />-->
				<AndroidEnvironment Include="Android\environment.conf" />
				<TrimmerRootDescriptor Include="Android\LinkerExclusions.xml" />
			</ItemGroup>

			<!-- Android resources. -->
			<!-- Shared resources. -->
			<ItemGroup>
				<AndroidResource Include="Android\Resources\values\Colors.xml" />
				<AndroidResource Include="Android\Resources\values\Styles.xml" />
				<AndroidResource Include="Android\Resources\drawable\splash.xml" />
				<AndroidResource Include="Android\Resources\xml\file_paths.xml" />
			</ItemGroup>

			<!-- ApplicationFlavor specific resources. -->
			<ItemGroup>
				<AndroidResource Include="Android\Resources_$(ApplicationFlavor)\drawable\*.*">
					<Link>Android\Resources\drawable\%(FileName)%(Extension)</Link>
				</AndroidResource>
				<AndroidResource Include="Android\Resources_$(ApplicationFlavor)\drawable-*dpi\*.*">
					<Link>Android\Resources\%(RecursiveDir)%(FileName)%(Extension)</Link>
				</AndroidResource>
				<AndroidResource Include="Android\Resources_$(ApplicationFlavor)\mipmap-*dpi\*.*">
					<Link>Android\Resources\%(RecursiveDir)%(FileName)%(Extension)</Link>
				</AndroidResource>
				<AndroidResource Include="Android\Resources_$(ApplicationFlavor)\values\Strings.xml">
					<Link>Android\Resources\values\Strings.xml</Link>
				</AndroidResource>
			</ItemGroup>

			<!-- Network security configuration -->
			<ItemGroup>
				<AndroidResource Include="Android\Resources\xml\network_security_config.xml" />
			</ItemGroup>
			<!-- Android Firebase configuration. -->
			<ItemGroup>
				<GoogleServicesJson Include="Android\Resources_$(ApplicationFlavor)\google-services.prod.json">
					<Link>google-services.json</Link>
				</GoogleServicesJson>
			</ItemGroup>
			<!-- Google api key -->
			<ItemGroup>
				<AndroidResource Include="Android\Resources\values\maps.xml" />
			</ItemGroup>
		</When>
		<When Condition="'$(TargetFramework)'=='net8.0-ios'">
			<PropertyGroup>
				<MtouchExtraArgs>$(MtouchExtraArgs) --setenv=MONO_GC_PARAMS=soft-heap-limit=512m,nursery-size=64m,evacuation-threshold=66,major=marksweep,concurrent-sweep</MtouchExtraArgs>
				<MtouchExtraArgs>$(MtouchExtraArgs) --xml=./iOS/LinkerExclusions.xml --linkskip=$(AssemblyName)</MtouchExtraArgs>
				<!-- See https://github.com/unoplatform/uno/issues/9430 for more details. -->
				<MtouchExtraArgs>$(MtouchExtraArgs) --registrar:static</MtouchExtraArgs>
				<!-- See https://github.com/xamarin/xamarin-macios/issues/14812 for more details. -->
				<MtouchExtraArgs>$(MtouchExtraArgs) --marshal-objectivec-exceptions:disable</MtouchExtraArgs>
				<CodesignEntitlements>iOS\Entitlements.plist</CodesignEntitlements>
				<!-- Workaround for clang++ error on iOS simulator on ARM base macOS system. See "https://github.com/dotnet/maui/issues/16778#issuecomment-1683343255" for more detail. -->
				<ForceSimulatorX64ArchitectureInIDE>true</ForceSimulatorX64ArchitectureInIDE>
			</PropertyGroup>
			<Choose>
				<When Condition="'$(Configuration)'=='Debug'">
					<PropertyGroup>
						<MtouchDebug>True</MtouchDebug>
						<MtouchExtraArgs>$(MtouchExtraArgs) --setenv=MONO_LOG_LEVEL=debug --setenv=MONO_LOG_MASK=gc</MtouchExtraArgs>
						<Optimize>False</Optimize>
						<ProvisioningType>manual</ProvisioningType>
					</PropertyGroup>
					<PropertyGroup Condition="'$(Platform)'=='iPhone'">
						<CodesignKey>iPhone Developer</CodesignKey>
					</PropertyGroup>
				</When>
				<When Condition="'$(Configuration)'=='Release'">
					<PropertyGroup>
						<BuildIpa>True</BuildIpa>
						<CodesignKey>iPhone Distribution</CodesignKey>
						<RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
						<OptimizePNGs>True</OptimizePNGs>
						<MtouchUseLlvm>True</MtouchUseLlvm>
						<MtouchUseThumb>True</MtouchUseThumb>
						<MtouchNoSymbolStrip>False</MtouchNoSymbolStrip>
						<Optimize>True</Optimize>
					</PropertyGroup>
				</When>
			</Choose>
			<ItemGroup>
				<PackageReference Include="Uno.Extensions.Logging.OSLog" Version="1.4.0" />
				<PackageReference Include="Xamarin.Firebase.iOS.Crashlytics" Version="8.10.0.3" />
				<PackageReference Include="Xamarin.Firebase.iOS.Analytics" Version="8.10.0.3" />
				<PackageReference Include="Xamarin.Firebase.iOS.PerformanceMonitoring" Version="8.10.0.3" />

				<!-- The latest version of SkiaSharp is required with Firebase. See https://github.com/unoplatform/uno/discussions/14009 for more details -->
				<PackageReference Include="SkiaSharp" Version="2.88.6" />
			</ItemGroup>
			<ItemGroup>
				<TrimmerRootDescriptor Include="iOS\LinkerExclusions.xml" />
			</ItemGroup>

			<!-- iOS resources -->
			<!-- Shared resources. -->
			<ItemGroup>
				<!--
				Removing this fix release build.
				<InterfaceDefinition Include="iOS\LaunchScreen.storyboard" />
				-->
				<BundleResource Include="iOS\Resources\en.lproj\AddToWallet.png" />
				<BundleResource Include="iOS\Resources\en.lproj\AddToWallet%403x.png" />
				<BundleResource Include="iOS\Resources\en.lproj\AddToWallet%402x.png" />
				<BundleResource Include="iOS\Resources\fr.lproj\AddToWallet.png.png" />
				<BundleResource Include="iOS\Resources\fr.lproj\AddToWallet%402x.png" />
				<BundleResource Include="iOS\Resources\fr.lproj\AddToWallet%403x.png" />
			</ItemGroup>
			<!-- ApplicationFlavor specific resources. -->
			<ItemGroup>
				<BundleResource Include="iOS\Resources_$(ApplicationFlavor)\Assets\**\*.*">
					<Link>iOS\Resources\Assets\%(RecursiveDir)%(FileName)%(Extension)</Link>
				</BundleResource>
				<BundleResource Include="iOS\Resources_$(ApplicationFlavor)\en.lproj\**\*.*">
					<Link>iOS\Resources\en.lproj\%(RecursiveDir)%(FileName)%(Extension) </Link>
				</BundleResource>
				<BundleResource Include="iOS\Resources_$(ApplicationFlavor)\fr.lproj\**\*.*">
					<Link>iOS\Resources\fr.lproj\%(RecursiveDir)%(FileName)%(Extension) </Link>
				</BundleResource>
				<ImageAsset Include="iOS\Resources_$(ApplicationFlavor)\Media.xcassets\**\*.*">
					<Link>Resources\Media.xcassets\%(RecursiveDir)%(FileName)%(Extension)</Link>
					<Visible>false</Visible>
				</ImageAsset>
			</ItemGroup>
			<!-- iOS Firebase configuration. -->
			<ItemGroup>
				<BundleResource Include="iOS\Resources_$(ApplicationFlavor)\GoogleService-Info.plist">
					<Link>GoogleService-Info.plist</Link>
				</BundleResource>
			</ItemGroup>
		</When>
	</Choose>
	<Import Project="..\ProjectName.Shared.Views\ProjectName.Shared.Views.projitems" Label="Shared" />

	<!-- Workaround to remove bitcode introduced by OneSignal. See https://github.com/OneSignal/OneSignal-DotNet-SDK/issues/27 for more details. -->
	<Import Project="..\..\..\BitcodeRemovalWorkaround.targets" Condition="'$(TargetFramework)' == 'net8.0-ios'" />
</Project>
```
2. Try to publish that project with the following command:
```shell
    cd ${{ parameters.pathToSrc }}/app/${{ parameters.solutionName }}.Mobile
    dotnet publish -f:net8.0-ios -c:Release /p:ApplicationFlavor=$(ApplicationFlavor) /p:ApplicationIdentifier=$(ApplicationIdentifier) /p:CodesignProvision=$(provisioningProfile.provisioningProfileUuid) /p:ArchiveOnBuild=true /bl:$(build.ArtifactStagingDirectory)/build-net8.0-ios.binlog "/p:InformationalVersion=$GITVERSION_INFORMATIONALVERSION" /p:IsLightBuild=false /p:FirebaseCrashlyticsUploadSymbolsEnabled=false
```

### Expected Behavior
The project should not hang while building.

### Actual Behavior
When it targets `net7.0-ios`, it hangs intermittently. See https://github.com/microsoft/azure-pipelines-agent/issues/4677.
When it targets `net8.0-ios`, it always hangs with the following message:

```
  ProjectName.Mobile -> /Users/runner/work/1/s/src/app/ProjectName.Mobile/bin/Release/net7.0-ios/ios-arm64/ProjectName.Mobile.dll
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/parasol.activities/0.40.5/lib/net6.0-ios16.1/Parasol.Activities.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/uno.fonts.fluent/2.3.0/lib/netstandard1.0/Uno.Fonts.Fluent.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/uno.fonts.roboto/2.2.2/lib/netstandard2.0/Uno.Fonts.Roboto.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/uno.toolkit.winui/4.2.22/lib/net6.0-ios15.4/Uno.Toolkit.WinUI.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/.nuget/packages/uno.toolkit.winui.material/4.2.22/lib/net6.0-ios15.4/Uno.Toolkit.WinUI.Material.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/work/1/s/src/app/ProjectName.Access/bin/Release/net7.0/ProjectName.Access.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/work/1/s/src/app/ProjectName.Business/bin/Release/net7.0/ProjectName.Business.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file '/Users/runner/work/1/s/src/app/ProjectName.Presentation/bin/Release/net7.0/ProjectName.Presentation.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7099/targets/Xamarin.Shared.Sdk.targets(1499,3): warning : The file 'bin/Release/net7.0-ios/ios-arm64/ProjectName.Mobile.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
  /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip
  /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip
  Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Optimizing assemblies for size. This process might take a while.
ILLINK : warning MT0079: The recommended Xcode version for Microsoft.iOS 16.4.7099 is Xcode 14.3 or later. The current Xcode version (found in /Applications/Xcode_14.2.app/Contents/Developer) is 14.2. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
  		
./iOS/LinkerExclusions.xml(9,3): warning IL2007: Could not resolve assembly 'ProjectName.DataAccess'. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
iOS/LinkerExclusions.xml(9,3): warning IL2007: Could not resolve assembly 'ProjectName.DataAccess'. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/.nuget/packages/xamarin.firebase.ios.core/8.10.0.3/lib/net6.0-ios15.4/Firebase.Core.resources/GoogleUtilitiesComponents.xcframework/ios-arm64/GoogleUtilitiesComponents.framework/GoogleUtilitiesComponents : warning MT7091: The framework /Users/runner/.nuget/packages/xamarin.firebase.ios.core/8.10.0.3/lib/net6.0-ios15.4/Firebase.Core.resources/GoogleUtilitiesComponents.xcframework/ios-arm64/GoogleUtilitiesComponents.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/FAnlytcs-8.9.1/FirebaseAnalytics-8.9.1/Frameworks/FirebaseAnalytics.xcframework/ios-arm64_armv7/FirebaseAnalytics.framework/FirebaseAnalytics : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/FAnlytcs-8.9.1/FirebaseAnalytics-8.9.1/Frameworks/FirebaseAnalytics.xcframework/ios-arm64_armv7/FirebaseAnalytics.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/GAppM-8.9.1/GoogleAppMeasurement-8.9.1/Frameworks/GoogleAppMeasurement.xcframework/ios-arm64_armv7/GoogleAppMeasurement.framework/GoogleAppMeasurement : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/GAppM-8.9.1/GoogleAppMeasurement-8.9.1/Frameworks/GoogleAppMeasurement.xcframework/ios-arm64_armv7/GoogleAppMeasurement.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Base/Frameworks/GoogleMapsBase.framework/GoogleMapsBase : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Base/Frameworks/GoogleMapsBase.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Maps/Frameworks/GoogleMapsCore.framework/GoogleMapsCore : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Maps/Frameworks/GoogleMapsCore.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
/Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Maps/Frameworks/GoogleMaps.framework/GoogleMaps : warning MT7091: The framework /Users/runner/Library/Caches/XamarinBuildDownload/GMps-6.0.1/Maps/Frameworks/GoogleMaps.framework is a framework of static libraries, and will not be copied to the app. [/Users/runner/work/1/s/src/app/ProjectName.Mobile/ProjectName.Mobile.csproj::TargetFramework=net7.0-ios]
##[error]The Operation will be canceled. The next steps may not contain expected logs.
##[error]The operation was canceled.
Finishing: Build project for Release - net7.0-ios
```

### Environment

It's mostly happening on our Azure DevOps CI, which is using Microsoft hosted agents. I've tried with the following:

1. `macos-12`: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md
2. `macos-13`: https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md

--- a/dotnet/targets/Xamarin.Shared.Sdk.targets
+++ b/dotnet/targets/Xamarin.Shared.Sdk.targets
@@ -1261,6 +1261,7 @@
 			Assemblies="@(_AssembliesToAOT)"
 			AOTCompilerPath="$(_XamarinAOTCompiler)"
 			InputDirectory="$(_AOTInputDirectory)"
+			MaxDegreeOfParallelism="$(AotCompileMaxDegreeOfParallelism)"
 			MinimumOSVersion="$(_MinimumOSVersion)"
 			OutputDirectory="$(_AOTOutputDirectory)\%(_AssembliesToAOT.Arch)"
 			SdkDevPath="$(_SdkDevPath)"

--- a/msbuild/Xamarin.MacDev.Tasks/Tasks/AOTCompile.cs
+++ b/msbuild/Xamarin.MacDev.Tasks/Tasks/AOTCompile.cs
@@ -16,7 +16,7 @@ using Xamarin.Utils;
 #nullable enable
 
 namespace Xamarin.MacDev.Tasks {
-	public class AOTCompile : XamarinTask, ITaskCallback, ICancelableTask {
+	public class AOTCompile : XamarinParallelTask, ITaskCallback, ICancelableTask {
 		public ITaskItem [] AotArguments { get; set; } = Array.Empty<ITaskItem> ();
 
 		[Required]
@@ -314,7 +314,7 @@ namespace Xamarin.MacDev.Tasks {
 				listOfArguments.Add (new (arguments, input));
 			}
 
-			Parallel.ForEach (listOfArguments, (arg) => {
+			ForEach (listOfArguments, (arg) => {
 				ExecuteAsync (AOTCompilerPath, arg.Arguments, sdkDevPath: SdkDevPath, showErrorIfFailure: false /* we show our own error below */)
 					.ContinueWith ((v) => {
 						if (v.Result.ExitCode != 0)

--- a/msbuild/Xamarin.MacDev.Tasks/Tasks/Codesign.cs
+++ b/msbuild/Xamarin.MacDev.Tasks/Tasks/Codesign.cs
@@ -17,7 +17,7 @@ using Xamarin.Utils;
 #nullable disable
 
 namespace Xamarin.MacDev.Tasks {
-	public class Codesign : XamarinTask, ITaskCallback, ICancelableTask {
+	public class Codesign : XamarinParallelTask, ITaskCallback, ICancelableTask {
 		const string ToolName = "codesign";
 		const string MacOSDirName = "MacOS";
 		const string CodeSignatureDirName = "_CodeSignature";
@@ -54,10 +54,6 @@ namespace Xamarin.MacDev.Tasks {
 
 		// Can also be specified per resource using the 'CodesignDeep' metadata (yes, the naming difference is correct and due to historical reasons)
 		public bool IsAppExtension { get; set; }
-
-		// How many codesign tasks we execute in parallel. Default is number of processors / 2.
-		// Contrary to many of the other codesigning properties, this can not be set per resource.
-		public string MaxDegreeOfParallelism { get; set; }
 
 		// Can also be specified per resource using the 'CodesignUseHardenedRuntime' metadata
 		public bool UseHardenedRuntime { get; set; }
@@ -103,16 +99,6 @@ namespace Xamarin.MacDev.Tasks {
 		string GetCodesignAllocate (ITaskItem item)
 		{
 			return GetNonEmptyStringOrFallback (item, "CodesignAllocate", CodesignAllocate, "CodesignAllocate", required: true);
-		}
-
-		int GetMaxDegreeOfParallelism ()
-		{
-			if (!string.IsNullOrEmpty (MaxDegreeOfParallelism)) {
-				if (int.TryParse (MaxDegreeOfParallelism, out var max))
-					return max;
-				Log.LogWarning (MSBStrings.W7121 /* Unable to parse the value '{0}' for the property 'MaxDegreeOfParallelism'. Falling back to the default value (number of processors / 2). */, MaxDegreeOfParallelism);
-			}
-			return Math.Max (Environment.ProcessorCount / 2, 1);
 		}
 
 		// 'sortedItems' is sorted by length of path, longest first.
@@ -482,7 +468,7 @@ namespace Xamarin.MacDev.Tasks {
 
 			for (var b = 0; b < buckets.Count; b++) {
 				var bucket = buckets [b];
-				Parallel.ForEach (bucket, new ParallelOptions { MaxDegreeOfParallelism = GetMaxDegreeOfParallelism () }, (item) => {
+				ForEach (bucket, (item) => {
 					Sign (item);
 
 					var files = GetCodesignedFiles (item.Item);

--- a/msbuild/Xamarin.MacDev.Tasks/Tasks/ScnTool.cs
+++ b/msbuild/Xamarin.MacDev.Tasks/Tasks/ScnTool.cs
@@ -10,7 +10,7 @@ using Xamarin.Messaging.Build.Client;
 using Xamarin.Utils;
 
 namespace Xamarin.MacDev.Tasks {
-	public class ScnTool : XamarinTask {
+	public class ScnTool : XamarinParallelTask {
 		#region Inputs
 
 		[Required]
@@ -95,7 +95,7 @@ namespace Xamarin.MacDev.Tasks {
 				bundleResources.Add (bundleResource);
 			}
 
-			Parallel.ForEach (listOfArguments, (arg) => {
+			ForEach (listOfArguments, (arg) => {
 				ExecuteAsync ("xcrun", arg.Arguments, sdkDevPath: SdkDevPath).Wait ();
 			});
 

--- a/msbuild/Xamarin.MacDev.Tasks/Tasks/SymbolStrip.cs
+++ b/msbuild/Xamarin.MacDev.Tasks/Tasks/SymbolStrip.cs
@@ -12,7 +12,7 @@ using Xamarin.Messaging.Build.Client;
 #nullable enable
 
 namespace Xamarin.MacDev.Tasks {
-	public class SymbolStrip : XamarinTask, ITaskCallback {
+	public class SymbolStrip : XamarinParallelTask, ITaskCallback {
 		#region Inputs
 
 		[Required]
@@ -60,7 +60,7 @@ namespace Xamarin.MacDev.Tasks {
 			if (ShouldExecuteRemotely ())
 				return new TaskRunner (SessionId, BuildEngine4).RunAsync (this).Result;
 
-			Parallel.ForEach (Executable, new ParallelOptions { MaxDegreeOfParallelism = Math.Max (Environment.ProcessorCount / 2, 1) }, (item) => {
+			ForEach (Executable, (item) => {
 				ExecuteStrip (item);
 			});
 

--- a/msbuild/Xamarin.MacDev.Tasks/Tasks/XamarinParallelTask.cs
+++ b/msbuild/Xamarin.MacDev.Tasks/Tasks/XamarinParallelTask.cs
@@ -1,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Xamarin.Localization.MSBuild;
+using Xamarin.Utils;
+
+#nullable enable
+
+namespace Xamarin.MacDev.Tasks {
+	public abstract class XamarinParallelTask : XamarinTask {
+		// How many tasks we execute in parallel. Default is number of processors / 2.
+		public string MaxDegreeOfParallelism { get; set; } = string.Empty;
+
+		int GetMaxDegreeOfParallelism ()
+		{
+			if (!string.IsNullOrEmpty (MaxDegreeOfParallelism)) {
+				if (int.TryParse (MaxDegreeOfParallelism, out var max))
+					return max;
+				Log.LogWarning (MSBStrings.W7121 /* Unable to parse the value '{0}' for the property 'MaxDegreeOfParallelism'. Falling back to the default value (number of processors / 2). */, MaxDegreeOfParallelism);
+			}
+			return Math.Max (Environment.ProcessorCount / 2, 1);
+		}
+
+		protected void ForEach<TSource> (IEnumerable<TSource> source, Action<TSource> body)
+		{
+			var options = new ParallelOptions { MaxDegreeOfParallelism = GetMaxDegreeOfParallelism () };
+			Parallel.ForEach (source, options, body);
+		}
+	}
+}

--- a/msbuild/Xamarin.Shared/Xamarin.Shared.targets
+++ b/msbuild/Xamarin.Shared/Xamarin.Shared.targets
@@ -1070,6 +1070,7 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 			Condition="'$(IsMacEnabled)' == 'true'"
 			DeviceSpecificIntermediateOutputPath="$(DeviceSpecificIntermediateOutputPath)"
 			IsWatchApp="$(IsWatchApp)"
+			MaxDegreeOfParallelism="$(ColladaMaxDegreeOfParallelism)"
 			ProjectDir="$(MSBuildProjectDirectory)"
 			ResourcePrefix="$(_ResourcePrefix)"
 			SdkPlatform="$(_SdkPlatform)"
@@ -2771,6 +2772,7 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 			Condition="'$(IsMacEnabled)' == 'true'"
 			Executable="$(_AppContainerDir)%(_NativeStripItems.Identity)"
 			Kind="%(_NativeStripItems.Kind)"
+			MaxDegreeOfParallelism="$(SymbolStripMaxDegreeOfParallelism)"
 			SymbolFile="%(_NativeStripItems.SymbolFile)"
 		/>
 

